### PR TITLE
Introduce runtime names for modules

### PR DESCRIPTION
**Pull request by tomchop**
Modules can now be identified by their `runtime_name` instead of their class names. This allows to disambiguate modules when there is more than one module of the same type in a recipe.

I didn't find a better variable name than `runtime_name`. WDYT?

### CODE REVIEW

**Comment by tomchop** [comment]
Moving to draft as I still need to add the kwarg to all existing modules. But feel free to take a look at the other changes.

File: dftimewolf/lib/module.py [hunk 1]
```diff
@@ -19,25 +19,31 @@ class BaseModule(object):
...
   """
 
-  def __init__(self, state, critical=False):
+  def __init__(self, state, name=None, critical=False):
     """Initialize a module.
 
     Args:
       state (DFTimewolfState): recipe state.
+      name (Optional[str]): The module's runtime name.
+      name (Optional[str]): A unique name for a specific instance of the module
```
Comment (kiddinn) [inline dftimewolf/lib/module.py:33]: why is name twice here?

**Review by kiddinn — COMMENTED** [review]
one tiny minor nit

File: dftimewolf/lib/module.py (hunk 1 above)
Comment (tomchop) [reply dftimewolf/lib/module.py:33]: I was overeager to document this new feature it seems.

**Review by berggren — APPROVED** [review]
LGTM

**Pull request merged**